### PR TITLE
[e2e] Expired fee is updated

### PR DESCRIPTION
**Pull request by W3stside**
Part of #116 

1. First start fresh from `localStorage` + wait for it to be populated by the frontend app
2. test pushes a fee expiring in the past and `expect`s it to be so
3. removes `localStorage` fee data
4. waits for it to be populated by the app
5. `expect`s the fee to expire in the future

### CODE REVIEW

File: cypress-custom/integration/fee.test.ts [hunk 1]
```diff
@@ -54,9 +54,53 @@ describe('Fetch and persist fee', () => {
 
   // TODO: not sure if it's easy to test this
   it('Re-fetched when it expires', () => {
+    const NETWORK = ChainId.RINKEBY.toString()
+    const TOKEN = 'ETH'
+    const KEY = 'redux_localstorage_simple_fee'
```
Comment (anxolin) [inline cypress-custom/integration/fee.test.ts:59]: refactor this with the foreach, and move up? sth like`const FEE_LOCAL_STORAGE_KEY` ?

File: cypress-custom/integration/fee.test.ts [hunk 2]
```diff
@@ -54,9 +54,53 @@ describe('Fetch and persist fee', () => {
...
+      .then($feeStorage => {
+        // we need to parse JSON
+        const feeStorage = JSON.parse($feeStorage)
+        const NOW = new Date()
+        const feeExpirationDate = new Date(feeStorage[NETWORK].ETH.fee.expirationDate)
+
+        // Here we expect the current saved expirationDate to be in the PAST
+        expect(NOW).to.be.greaterThan(feeExpirationDate)
+      })
+      // WHEN: When the fee quote expires, we refetch the fee
```
Comment (anxolin) [inline cypress-custom/integration/fee.test.ts:92]: uber mega nit. On blank line between the GIVEN and the WHEN?

File: cypress-custom/integration/fee.test.ts [hunk 3]
```diff
@@ -54,9 +54,53 @@ describe('Fetch and persist fee', () => {
...
+      }
+    })
     // GIVEN: A fee is present in the local storage
-    // WHEN: When the fee quote expires, we refetch the fee
-    // THEN: We get another quote
+    // Set expiring fee in localStorage
+    cy.get<Storage>('@localStorage')
+      .then($storage => {
+        // set time in the past
+        $storage.setItem(KEY, VALUE)
```
Comment (anxolin) [inline cypress-custom/integration/fee.test.ts:77]: why separating the VALUE from the given? 

don't fix, I'll give a try on making this more slim in a bit

File: cypress-custom/integration/fee.test.ts [hunk 4]
```diff
@@ -54,9 +54,53 @@ describe('Fetch and persist fee', () => {
...
+    const NETWORK = ChainId.RINKEBY.toString()
+    const TOKEN = 'ETH'
+    const KEY = 'redux_localstorage_simple_fee'
+    const VALUE = JSON.stringify({
+      [NETWORK]: {
+        [TOKEN]: {
+          fee: {
+            minimalFee: '777',
+            feeRatio: 777,
+            expirationDate: new Date(Date.now() - 200000).toISOString()
```
Comment (anxolin) [inline cypress-custom/integration/fee.test.ts:66]: my initial thinking was to set some time in the future always and advance time. Maybe this helps https://docs.cypress.io/api/commands/tick.html#Syntax

Comment (anxolin) [reply cypress-custom/integration/fee.test.ts:66]: Reason is, we want to really reproduce a workflow where the user has a key, come back some days later, and then we make sure he uses the good fee

**Review by alfetopito — APPROVED** [review]
LGTM

File: cypress-custom/integration/fee.test.ts (hunk 4 above)
Comment (W3stside) [reply cypress-custom/integration/fee.test.ts:66]: Hmm could be a good idea, didn't see tick before. Will check out tomorrow !

**Comment by anxolin** [comment]
Proposal of PR on top of this one https://github.com/gnosis/gp-swap-ui/pull/121 for adding a new command. Also I changed a bit the implementation of the test

**Review by anxolin — APPROVED** [review]

File: cypress-custom/integration/fee.test.ts (hunk 4 above)
Comment (anxolin) [reply cypress-custom/integration/fee.test.ts:69]: Moved to an issue so we can merge this PR https://github.com/gnosis/gp-swap-ui/issues/125

**Pull request merged**